In [1]:
from keys import FTX_API_KEY, FTX_API_SECRET
from datetime import datetime
import time
import hmac
import pandas as pd
from requests import Request, Session, Response
from typing import Optional, Dict, Any, List
from utility_classes.historical_data_processor import FTXDataProcessor
import dateutil.parser as dp


## all non-expired futures tickers

In [7]:
import json

acc = FTXDataProcessor(api_key=FTX_API_KEY, api_secret=FTX_API_SECRET)
current_futures = acc.get_all_futures_tickers()
for i in current_futures:
    print(i)


109
1INCH
CUSDT
BTMX
CAKE
YFII
COMP
ALGO
DMG
UNISWAP
LUNA
MATIC
ONT
AMPL
BNB
EXCH
SNX
BCH
LINA
LEO
TRU
FLOW
DOT
BAL
AVAX
AAVE
EGLD
KNC
MID
LRC
SRM
HOLY
FIDA
BAO
LTC
SXP
QTUM
ZIL
DEFI
TRX
BRZ
THETA
RSR
YFI
XLM
RUNE
FLM
SHIT
ETC
KAVA
ZRX
ENJ
REEF
ETH
SECO
DRGN
SUSHI
HNT
BAT
CHZ
OMG
BSV
ROOK
MAPS
USDT
KIN
EOS
ADA
NEAR
PERP
XRP
CRV
FTT
MKR
BAND
ATOM
ALT
AR
WAVES
CREAM
HT
OKB
TOMO
KSM
BTC
RAY
BNT
OXY
SOL
BTT
XAUT
GRT
PAXG
MTA
LINK
VET
ZEC
REN
UNI
NEO
PRIV
ALPHA
XTZ
FTM
XMR
TRYB
BADGER
DOGE
DASH
FIL


## getting all expired futures

In [3]:
expired_futures = acc._get('expired_futures')
for ticker in expired_futures: 
    if ticker['underlying'] == 'ETH':
        print(ticker['name'], ticker['expiryDescription'], ticker['expiry'])


ETH-1225 December 2020 2020-12-25T03:00:00+00:00
ETH-0925 September 2020 2020-09-25T03:00:00+00:00
ETH-0626 June 2020 2020-06-26T03:00:00+00:00
ETH-20200327 March 2020 2020-03-27T03:00:00+00:00
ETH-20191227 December 2019 2019-12-27T03:00:00+00:00
ETH-20190927 September 2019 2019-09-27T03:00:00+00:00
ETH-20190628 June 2019 2019-06-28T03:00:00+00:00
ETH-20190329 March 2019 2019-03-29T03:00:00+00:00


## all past futures expiration dates
### to get past data requires correct date input 
### only futures data after 20190628 is available

In [4]:
acc.get_expired_futures_dates()


expiration date: 1225 | Date December 2020
expiration date: 0925 | Date September 2020
expiration date: 0626 | Date June 2020
expiration date: 20200327 | Date March 2020
expiration date: 20191227 | Date December 2019
expiration date: 20190927 | Date September 2019
expiration date: 20190628 | Date June 2019
expiration date: 20190329 | Date March 2019


## specific expired futures data 


In [5]:
eth_hist_futures = acc.get_expired_futures_OHCL(market = 'ETH-20190927', resolution = 3600)
print(eth_hist_futures)

Adding 1648 candles with start time 2019-09-26 23:00:00
Adding 1 candles with start time 2019-07-20 08:00:00
                      timestamp    open    high     low   close     volume  \
0     2019-07-20T12:00:00+00:00  227.76  228.38  227.41  227.62    0.00000   
1     2019-07-20T13:00:00+00:00  227.62  230.33  225.93  227.06    0.00000   
2     2019-07-20T14:00:00+00:00  227.06  228.91  227.01  228.80    0.00000   
3     2019-07-20T15:00:00+00:00  228.80  232.33  228.01  231.90    0.00000   
4     2019-07-20T16:00:00+00:00  231.90  236.97  231.49  235.48    0.00000   
...                         ...     ...     ...     ...     ...        ...   
1643  2019-09-26T23:00:00+00:00  166.77  167.73  165.82  166.10    0.00000   
1644  2019-09-27T00:00:00+00:00  166.10  166.71  164.19  164.34  198.98362   
1645  2019-09-27T01:00:00+00:00  164.26  165.89  163.83  165.84  339.72944   
1646  2019-09-27T02:00:00+00:00  165.81  165.95  164.33  165.05    0.00000   
1647  2019-09-27T03:00:00+00:00  